### DISCUSSION TITLE
Is reverse engineering a problem if you write whole game in Swift?

### DISCUSSION THREAD

**Discussion by Qws** [migueldeicaza/SwiftGodot#444] [General]
Apparently it takes almost no effort to reverse engineer a Godot game relying on GDscript, but also a very little effort to reverse engineer a .NET Godot game.

I have interest in SwiftGodot, simply because I want to learn Swift and I think a game engine that doesn't use garbage collector is a very good thing. But I'm curious how reverse engineering state is with SwiftGodot? Is it harder or just as easy? Or does it depend on the platform?

**Comment by migueldeicaza** [comment]
languages that compile to native code like swift or c are harder to reverse engineer than interpreted languages or systems like c# and Java that use bytecodes that are both easy to decompile and have a good ecosystem of decompiling tools.

But it is not impossible, merely harder.   Swift is harder than C in a few dimensions, existing professional decompilers for c/c++ have not been updated to deal with swift-isism, so it is even more obnoxious. 

But in the end, decompilation for all languages is possible, it is just a matter of how hard someone really wants to get at the data and how much they are willing to invest in it.

overall, I think that swift and other compiled languages like c/c++ will be beyond the means of casual observers that could have cracked trivially a gdscript or c# program, but it is by no means impenetrable.
